### DISCUSSION TITLE
@tailwindcss/forms with @headlessui/react@v2

### DISCUSSION THREAD

**Discussion by lishaduck** [tailwindlabs/headlessui#3204] [Help]
Hello! I'm upgrading a [website](https://github.com/PHS-TSA/webmaster-23-24) to `@headlessui/react` v2. I use `@tailwindcss/forms`. The new method of checkboxing-ing uses `div`s, rather than inputs of type `checkbox`, so the checkbox styling from `@tailwindcss/forms` no longer works. Is there a new way of applying the styles?
I looked, and it seems that there isn't a supported way to emulate states, though I can apply the `form-checkbox` class to get some styles.

Example:
```tsx
<Checkbox name="somename" className="form-checkbox" required />
```

If you hit check, it _is_ selected, but it doesn't show it.
Is there a way to work around this?

> [!NOTE]
> I am using Preact via Fresh, but that doesn't seem to be related.

**Comment by tobi1220** [comment]
Not an answer but just a quick heads-up, I'm pretty sure HeadlessUI doesn't support `required` on `<Checkbox />`

**Reply by lishaduck** [reply]
Oh, yeah. That's true as well :)
All the more reason to use `<Input />` instead.
